sys/net/network_layer/sixlowpan/mac.c

```diff
@@ -168,17 +168,13 @@ static void *recv_ieee802154_frame(void *arg)
     return NULL;
 }
 
-void set_ieee802154_fcf_values(ieee802154_frame_t *frame, uint8_t dest_mode,
-                               uint8_t src_mode)
+void set_ieee802154_fcf_values(ieee802154_frame_t *frame, uint16_t dest_mode,
+                               uint16_t src_mode)
 {
-    frame->fcf.frame_type = IEEE_802154_DATA_FRAME;
-    frame->fcf.sec_enb = 0;
-    frame->fcf.frame_pend = 0;
-    frame->fcf.ack_req = 0;
-    frame->fcf.panid_comp = (frame->dest_pan_id == frame->src_pan_id);
-    frame->fcf.frame_ver = 0;
-    frame->fcf.src_addr_m = src_mode;
-    frame->fcf.dest_addr_m = dest_mode;
+    frame->fcf = (IEEE_802154_FCF_TYPE_DATA |
+                  ((frame->dest_pan_id == frame->src_pan_id) << IEEE_802154_FCF_PANID_COMP_POS) |
+                  dest_mode |
+                  src_mode);
```

Comment:
Not 100% :-) But I had some cases in the past where the compiler was not smart enough to optimize, that's why I got used to doing these kind of assignments in one step...